### ISSUE TITLE
V3.9.0 Tween resume() seems to fail if called too soon after pause()?

### ISSUE THREAD

**Issue by thorie7912** [photonstorm/phaser#3721]
I'm not sure if this is a bug or I'm just missing some basic concept. I'm trying to tween a sprite every time a keypress happens, unless the sprite is already moving. The code below works fine if I press the `D` key repeatedly but slowly. However, if I hold down the `D` key (and it may depend on your own operating system's key repeat speed settings), then eventually everything gets "stuck" into a state where the tween stops working. Looking at the logs, it seems like `tween.resume()` was called but the `onComplete()` never executed.

My "theory" is that `tween.pause()` actually takes some time, and so if `tween.resume()` is called too early, then it will not actually resume. I also tried checking `tween.isPaused()` during debugging before resume, but isPaused was `false` yet resume still seemed to not work properly.

Any ideas what I am doing wrong and how to avoid getting "stuck" where the tween doesn't work on every D keypress anymore?

```
<!DOCTYPE html>
<html>
    <head>
        <script src="//cdn.jsdelivr.net/npm/phaser@3.9.0/dist/phaser.js"></script>
    </head>
    <body>
        Test
        <script>
var config = {
    type: Phaser.CANVAS,
    width: 800,
    height: 600,
    backgroundColor: '#2d2d2d',
    scene: {
        preload: preload,
        create: create
    }
};

var game = new Phaser.Game(config);

function preload() {
    this.load.setBaseURL('https://labs.phaser.io');
    this.load.image('arrow', 'assets/sprites/arrow.png');
}

function create() {
    var marker = this.add.image(100, 100, 'arrow').setAlpha(0.3);
    image = this.add.image(100, 100, 'arrow');

    tween = this.tweens.add({
        targets: image,
        x: 200,
        duration: 500,
        paused: true,
        completeDelay: 0,
        onComplete: moveComplete
    });
    
    this.input.keyboard.on('keydown_D', movePlayer);
    alreadyMoving = false;
    
    console.log('Pressing the D key will tween the arrow once. The arrow position is\
    will be reset at the end of the movement. This works fine if you press the D key\
    repeatedly, but slowly. However, if you press it rapidly or hold down the D key\
    then the behavior will get "stuck" and the tween will no longer work.');
    
    debug = false;
}

function movePlayer()
{
    if (debug) console.log('movePlayer');
    if (!alreadyMoving) {
        alreadyMoving = true;
        tween.resume();
    } else {
        if (debug) console.log('cannot move yet')
    }
}

function moveComplete(tween, target)
{
    if (debug) console.log('moveComplete', tween);
    image.x = 100;
    
    // my bug theory is that pause() takes some time before resume() can be called again
    // if i set `alreadyMoving=false` in a setTimeout ~20ms after the pause()
    // then the arrow doesn't get "stuck"
    
    tween.pause();
    alreadyMoving = false;
}

</script>
</body>
</html>
```

**Comment by thorie7912** [comment]
Looking at the phaser source, I see:

```
isPaused: function ()
    {
        return (this.state === TWEEN_CONST.PAUSED);
    }
```
and
```
pause: function ()
    {
        if (this.state === TWEEN_CONST.PAUSED)
        {
            return;
        }

        this.paused = true;

        this._pausedState = this.state;

        this.state = TWEEN_CONST.PAUSED;

        return this;
    },
```
So, I theorized that there is an async issue calling `pause()` because it will only set `this.state = TWEEN_CONST.PAUSED` at the end of the call which has a var that gets overwritten by `resume()` setting a new state while the `pause()` is still being executed.

**Comment by thorie7912** [comment]
I was able to fix my code by changing the check inside `movePlayer()` to use `if (!tween.isPlaying()) {` instead of `if (!alreadyMoving)` and removing the `tween.pause()` and `alreadyMoving = false` from the `moveComplete` function. I think the basic issue was calling `tween.pause()` inside of the onComplete method would put things into a bad state.

I'll close this issue since I was able to fix it and it seems fine to me now, though I don't know if there is actually still a bug somewhere.
